### PR TITLE
[text analytics] Fixed Swagger validation issues for v3.1-preview.4

### PR DESCRIPTION
**Pull request by abhahn**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [ ] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: custom-words.txt
Comment (nschonni) [inline custom-words.txt:2064]: Some of these are actual typos that should be fixed or ignored for a specific version rather than globally ignored

Comment (abhahn) [reply custom-words.txt]: Thanks, I just made some updates to both files.  Let me know if you notice any other issues here.

Comment (nschonni) [reply custom-words.txt]: Nice! I'm trying to update the instructions over in https://github.com/Azure/azure-rest-api-specs/pull/12624

Comment (abhahn) [reply custom-words.txt]: Thanks for sharing!  Based on this, I'm wondering if I should instead add an override for Text Analytics in cSpell.json, since most of what I added are acronyms and not real words.

Comment (nschonni) [reply custom-words.txt]: I'm not an MS employee, I just setup the linting years ago and have been poking around to see the state it's in recently, so go with whatever the MS folks agree with😄

Comment (abhahn) [reply custom-words.txt]: Ok, I think I will leave it as-is for now.  This PR is meant for solving some CI issues and its contents will undergo a final review before everything is eventually merged to master.  :)

File: specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.1-preview.4/TextAnalytics.json
Comment (ctufts) [inline specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.1-preview.4/TextAnalytics.json:2750]: @abhahn  this doesn't have to be ["Default"] due to type: array? Or is this basically specifying that out of the enums, use this as the default value in the array?

Comment (abhahn) [reply specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.1-preview.4/TextAnalytics.json:2750]: Yeah, to be completely honest I don't really understand why this works but setting `"default": ["Default"]` doesn't.  This is the only way that I could get both Swagger model validation and Swagger linting to pass.  I think it may have the consequence of leaving the endpoint with no default value from the standpoint of APIM, which may be a problem.  If I understand correctly, I think `x-ms-client-default` will result in the generated SDK methods having a default value.  

Of the things I tried, here are the combinations that will fail one step or the other:  `"default": ["Default"]` (fails linting), `"default": "Default"` (fails model validation), and `"x-ms-client-default": ["Default"]` (fails linting).

I think that `"default": ["Default"]` *should* be the correct way (even though it is failing) because it is explicitly shown as the way to set default values for enums in the [official OpenApi 2.0 docs](https://swagger.io/docs/specification/2-0/describing-parameters/).  Unfortunately this page doesn't have anchors, but if you Ctrl+F you can find the following text for the example:  "You can also specify the default array that the server will use if this parameter is omitted"

Comment (thediris) [reply specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.1-preview.4/TextAnalytics.json:2750]: good catch, I didn't see the bit in the doc that allows you to omit the brackets

**Review by thediris — CHANGES_REQUESTED** [review]

File: specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.1-preview.4/TextAnalytics.json
Comment (abhahn) [reply specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.1-preview.4/TextAnalytics.json]: Thanks, I just updated this.  Let me know if you notice any other issues.

File: specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.1-preview.4/TextAnalytics.json
Comment (abhahn) [reply specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.1-preview.4/TextAnalytics.json:2750]: @tjprescott , just wondering if you could help us to understand why using an array to represent the default value using the `default` property won't work here.  I'd like to resolve this as soon as possible so we can merge the main PR for v3.1-preview.4.

**Comment by abhahn** [comment]
@anuchandy , could you please have a look at this PR when you have a chance?  I actually have a question in the comments [above](https://github.com/Azure/azure-rest-api-specs/pull/12660#discussion_r565664148) regarding the use of the `default` property as well, and was hoping someone could help answer this before merging.

File: specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.1-preview.4/TextAnalytics.json
Comment (anuchandy) [reply specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.1-preview.4/TextAnalytics.json:2751]: @abhahn and I discussed this offline. 

Given the property/parameter is an array type, like Abby pointed the swagger should represent the Default in an array format. But today only the property/parameter with schemas - string, integer, long, float, double, boolean is supported.

We kicked off a discussion with the AutoRest core team to evaluate enabling array schema for `x-ms-client-default` extension.

It may take some time for AutoRest core to evaluate and decide on the this feature request.

Meantime, we should not use this extension for array cases, as it cannot be translated to meaningful representation in autorest or any languages.

Comment (anuchandy) [reply specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.1-preview.4/TextAnalytics.json]: thanks! Laurent @lmazuel for all offline help on this.

**Review by anuchandy — APPROVED** [review]

**Pull request merged**
